source/grid/manifold.cc

```diff
@@ -659,16 +654,20 @@ add_new_points (const std::vector<Point<spacedim> > &surrounding_points,
           if (new_point[d] < 0)
             new_point[d] += periodicity[d];
 
-      new_points.push_back(project_to_manifold(surrounding_points, new_point));
+      // TODO should this use surrounding_points_start or surrounding_points?
+      // The older version used surrounding_points
+      new_points[row] = project_to_manifold(make_array_view(surrounding_points.begin(),
+                                                            surrounding_points.end()),
+                                            new_point);
```

Comment:
I would use `surrounding_points_start` but it might be that `project_to_manifold` does the necessary adjustments internally. Do we have tests that trigger on one or the other?

Reply:
I believe that the answer is 'yes' and our current implementation is wrong. At the same time, we do not use the surrounding points in `FlatManifold::project_to_manifold` (which is another bug in itself; this function should not be the identity function for `structdim != spacedim`) so this is really a separate problem.